crates/pallet-feeds/src/feed_processor.rs

```diff
@@ -17,13 +17,13 @@
 
 use sp_runtime::{DispatchError, DispatchResult};
 use sp_std::{vec, vec::Vec};
-use subspace_core_primitives::{crypto, Sha256Hash};
+use subspace_core_primitives::crypto;
 
 /// Object mapping that points to an object in a block
 #[derive(Debug)]
 pub struct FeedObjectMapping {
     /// Key scoped to the feed
-    pub key: Sha256Hash,
+    pub key: Vec<u8>,
```

Comment:
I was trying to understand, but can't see why this needs to be a vector now.

Reply:
Key as [u8; 32] forces the feed processor impl to use a some hash function and we do sha256 again while namespacing to feed as well. 

NoValidation Feed will still use sha256 twice but for other current Validation feeds, we are just giving back encoded(block_height, hash). Using Vec makes would be make sense to me since I assume NoValidation feeds would be far less that validation feeds.

I´m still not hapy with this change and have slept on it thinking of a better approach to scope the key without further allocation. Thoughts?

Reply:
You don't have to hash it, you just need to have 256-bit value. If you have let's say, 32-bit number, you can still fill first 32 bits with it and the rest with zeroes. It will be feed processor author's decision whether to hash it or make fixed size in some other way.

BTW, I feel like we need a special case for content-addressable content since we'd really like to make it the same regardless of whether it is stored with object store or any of the feeds, the hash should always be the same (useful for seamless IPFS integration for instance).

Reply:
> You don't have to hash it, you just need to have 256-bit value. 

what about the case when there are ore than 32 bytes like for example (block_height, block_hash) ?

Reply:
> BTW, I feel like we need a special case for content-addressable content since we'd really like to make it the same regardless of whether it is stored with object store or any of the feeds, the hash should always be the same (useful for seamless IPFS integration for instance).

Sure. we can move it as part of the object store maybe. Adding it part of the feeds means we have to make exceptions. Maybe part of different PR. WDYT ?

Reply:
> what about the case when there are ore than 32 bytes like for example (block_height, block_hash) ?

Then do hashing

> Sure. we can move it as part of the object store maybe. Adding it part of the feeds means we have to make exceptions. Maybe part of different PR. WDYT ?

Definitely. Not everything I say in comments needs to be acted upon immediately, this is something I noticed we need to do eventually. I think mapping should be on the processor layer in a sense of providing region of the payload, but feeds need to then read and actually hash it and create final mapping without namespacing. This is a separate special API and not a part of this PR.

Reply:
> Then do hashing

This is precisely what I was trying to avoid. Doing the double hashing on runtime. I will revert that chage

Reply:
> but feeds need to then read and actually hash it and create final mapping without namespacing. This is a separate special API and not a part of this PR.

I'm not sure about that. That would make feeds read the object from with in the object and then hash it. Wouldn't adding this logic as part of pallet- objects make more sense? That pallet could be like ipfs

Reply:
> Wouldn't adding this logic as part of pallet- objects make more sense?

We have object store pallet and it does work like that, but why wouldn't pallet feeds support it too?